gems/aws-sdk-core/lib/aws-sdk-core/deprecations.rb

```diff
@@ -35,33 +35,39 @@ module Aws
   # @api private
   module Deprecations
 
-    # @param [Symbol] method_name The name of the deprecated method.
+    # @param [Symbol] method The name of the deprecated method.
     #
     # @option options [String] :message The warning message to issue
     #   when the deprecated method is called.
     #
-    # @option options [Symbol] :use The name of an use
-    #   method that should be used.
+    # @option options [String] :use The name of a method that should be used.
     #
-    def deprecated(method_name, options = {})
+    # @option options [String] :version The version that will remove the
+    #   deprecated method.
+    #
+    def deprecated(method, options = {})
 
       deprecation_msg = options[:message] || begin
```

Comment:
Can we have this PR in 2 steps?
1. add stronger deprecation waring for those methods with a version number/deadline when they will be removed
2. perform removal after reaching the deadline

The `:message` parameter would be able to help provide additional messagings

Reply:
I agree. I arbitrarily chose 3.75 as the version. I will update the PR.